pymc_marketing/mmm/fourier.py

```diff
@@ -0,0 +1,616 @@
+
+    training_dates = dates[:52 * 2]
+    testing_dates = dates[52 * 2:]
+
+    yearly = YearlyFourier(n_order=3)
+
+    coords = {
+        "date": training_dates,
+    }
+    with pm.Model(coords=coords) as model:
+        dayofyear = pm.Data(
+            "dayofyear",
+            training_dates.dayofyear.to_numpy(),
+            dims="date",
+        )
+
+        trend = pm.Deterministic(
+            "trend",
+            yearly.apply(dayofyear),
+            dims="date",
+        )
+
+        idata = pm.sample_prior_predictive().prior
+
+    with model:
+        pm.set_data(
+            {"dayofyear": testing_dates.dayofyear.to_numpy()},
+            coords={"date": testing_dates},
+        )
+
+        out_of_sample = pm.sample_posterior_predictive(
+            idata,
+            var_names=["trend"],
+        ).posterior_predictive["trend"]
+
+
+Use yearly and monthly fourier seasonality together.
+
+By default, the prefix of the fourier seasonality is set to "fourier". However,
+the prefix can be changed upon initialization in order to avoid variable name
+conflicts.
+
+.. code-block:: python
+
+    from pymc_marketing.mmm import (
+        MonthlyFourier,
+        YearlyFourier,
+    )
+
+    yearly = YearlyFourier(n_order=6, prefix="yearly")
+    monthly = MonthlyFourier(n_order=3, prefix="monthly")
+
+    dates = pd.date_range("2023-01-01", periods=52, freq="W-MON")
+    dayofyear = dates.dayofyear.to_numpy()
+
+    coords = {
+        "date": dates,
+    }
+
+    with pm.Model(coords=coords) as model:
+        yearly_trend = yearly.apply(dayofyear)
+        monthly_trend = monthly.apply(dayofyear)
+
+        trend = pm.Deterministic(
+            "trend",
+            yearly_trend + monthly_trend,
+            dims="date",
+        )
+
+    with model:
+        prior_samples = pm.sample_prior_predictive().prior
+
+"""
+
+from typing import Any
+
+import arviz as az
+import matplotlib.pyplot as plt
+import numpy as np
+import numpy.typing as npt
+import pymc as pm
+import pytensor.tensor as pt
+import xarray as xr
+
+from pymc_marketing.constants import DAYS_IN_MONTH, DAYS_IN_YEAR
+from pymc_marketing.mmm.plot import (
+    plot_hdi,
+    plot_samples,
+)
+from pymc_marketing.prior import Prior, create_dim_handler
+
+
+def generate_fourier_modes(
+    periods: pt.TensorLike,
+    n_order: int,
+) -> pt.TensorVariable:
+    """Create fourier modes for a given period.
+
+    Parameters
+    ----------
+    periods : pt.TensorLike
+        Periods to generate fourier modes for.
+    n_order : int
+        Number of fourier modes to generate.
+
+    Returns
+    -------
+    pt.TensorVariable
+        Fourier modes.
+
+    """
+    multiples = pt.arange(1, n_order + 1)
+    x = 2 * pt.pi * periods
+
+    values = x[:, None] * multiples
+
+    return pt.concatenate(
+        [
+            pt.sin(values),
+            pt.cos(values),
+        ],
+        axis=1,
+    )
```

Comment:
This changes order of the coefficients. Is that an issue?

Reply:
At first sight, it looks fine!